### PR TITLE
[pickers] Use the new `ownerState` object on the `<PickersTextField />` component

### PR DESCRIPTION
**Pull request by flaviendelangle**
Part of #14475

I forgot this one :grimacing:

### CODE REVIEW

File: packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.types.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export interface PickersArrowSwitcherOwnerState extends PickerOwnerState {
   /**
    * If `true`, this button should be hidden.
    */
-  hidden: boolean;
+  isButtonHidden: boolean;
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/components/PickersArrowSwitcher/PickersArrowSwitcher.types.tsx:49]: I renamed that one to be consistent on the prefixing

File: packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
+import { usePickerTextFieldOwnerState } from '../usePickerTextFieldOwnerState';
 
 export interface PickersFilledInputProps extends PickersInputBaseProps {
   disableUnderline?: boolean;
   hiddenLabel?: boolean;
 }
 
+export interface PickersFilledInputOwnerState extends PickersTextFieldOwnerState {
+  /**
+   * If `true`, the input  have an underline, `false` otherwise.
```
Comment (michelengelen) [inline packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:27]:
```suggestion
   * If `true`, the input has an underline, if `false` it has not.
```

File: packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx [hunk 3]
```diff
@@ -120,13 +129,13 @@ const PickersFilledInputRoot = styled(PickersInputBaseRoot, {
         },
       },
       {
-        props: ({ startAdornment }: OwnerStateType) => !!startAdornment,
+        props: { isInputAdornedStart: true },
```
Comment (michelengelen) [inline packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:132]: Super nitty:

```suggestion
        props: { isInputAdornedAtStart: true },
```

Comment (michelengelen) [reply packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:132]: Absolutely "non-blocking" 🫣

File: packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx [hunk 4]
```diff
@@ -120,13 +129,13 @@ const PickersFilledInputRoot = styled(PickersInputBaseRoot, {
...
         style: {
           paddingLeft: 12,
         },
       },
       {
-        props: ({ endAdornment }: OwnerStateType) => !!endAdornment,
+        props: { isInputAdornedEnd: true },
```
Comment (michelengelen) [inline packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:138]:
```suggestion
        props: { isInputAdornedAtEnd: true },
```

Same as above

File: packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx [hunk 5]
```diff
@@ -21,14 +21,16 @@ import {
...
+import { PickersTextFieldOwnerState } from '../PickersTextField.types';
 
 const round = (value: number) => Math.round(value * 1e5) / 1e5;
 
 export const PickersInputBaseRoot = styled('div', {
   name: 'MuiPickersInputBase',
   slot: 'Root',
   overridesResolver: (props, styles) => styles.root,
-})<{ ownerState: OwnerStateType }>(({ theme }) => ({
+})<{ ownerState: PickersTextFieldOwnerState }>(({ theme }) => ({
```
Comment (michelengelen) [inline packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx:33]: Just noticed this:
Do we intentionally use "TextField" here to show the meaning of the overall concept and "input" elsewhere to target the actual element?

**Review by michelengelen — APPROVED** [review]
Only nits again! LGTM otherwise! 💪🏻

**Review by LukasTy — COMMENTED** [review]
Great work! 🎉 
Leaving some nitpicks. 🙈

File: packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx (hunk 3 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:132]: I went for `hasStartAdornment` which was proposed by Lukas :+1:

File: packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/PickersTextField/PickersFilledInput/PickersFilledInput.tsx:27]: I rephrased to be coherent with the other properties

File: packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx (hunk 5 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx]: The Text Field is the component that contains the label, the error message etc... AND the input.

It's not a nomenclature I love tbh, but I tried to stay close from the core logic here to smoothen the migration :+1:

**Pull request merged**